packages/babel-generator/src/generators/typescript.ts

```diff
@@ -589,15 +589,15 @@ export function TSModuleDeclaration(
   this: Printer,
   node: t.TSModuleDeclaration,
 ) {
-  const { declare, id } = node;
+  const { declare, id, kind } = node;
 
   if (declare) {
     this.word("declare");
     this.space();
   }
 
   if (!node.global) {
-    this.word(id.type === "Identifier" ? "namespace" : "module");
+    this.word(kind ?? (id.type === "Identifier" ? "namespace" : "module"));
```

Comment:
Even if we add the kind to the AST I would prefer to not do this change, since we currently always emit "good" code and there is no need to emit `module` (similarly to how we, for example, always emit semicolons even if they are absent in the input code).